src/NCronJob/Scheduler/JobWorker.cs

```diff
@@ -193,62 +195,46 @@ public void ScheduleJob(JobDefinition job)
         {
             LogNextJobRun(job.Type, nextRunTime.Value);  // todo: log by subscribing to OnStateChanged => JobStateType.Scheduled
             var run = JobRun.Create(timeProvider, observer.Report, job, nextRunTime.Value);
-            var jobQueue = jobQueueManager.GetOrAddQueue(job.JobFullName);
             jobQueue.Enqueue(run, (nextRunTime.Value, (int)run.Priority));
             run.NotifyStateChange(JobStateType.Scheduled);
         }
     }
 
     public void RemoveJobByName(string jobName)
     {
-        RemoveJobRunByName(jobName);
-        registry.RemoveByName(jobName);
+        RemoveJob(
+            () => registry.FindJobDefinition(jobName),
+            () => registry.RemoveByName(jobName));
+    }
+
+    public void RemoveJobByType(Type type)
+    {
+        RemoveJob(
+            () => registry.FindFirstJobDefinition(type),
+            () => registry.RemoveByType(type));
     }
 
-    public void RemoveJobType(Type type)
+    private void RemoveJob(
```

Comment:
Instead of the `Func<JobDefinition?>` why not accepting a `string jobFullName`?  and move the responsibility towards RemoveJobByXXX.

Reply:
Fixed